### PR TITLE
Create activity model, simple view, activity manager, and test for ba…

### PR DESCRIPTION
**Pull request by elsa-bismuth**
# **🔄 Create Activity Model, View, and Tests**

## **♻️ Current Situation & Problem**
This PR introduces the **Activity model, Activity Manager, Activity View, and corresponding tests**.  
Previously, there was no structured way to **log, track, and manage** kids' physical activity in the app.  

This PR implements:
- **A clean and structured Activity model** that stores necessary activity data.
- **An Activity Manager** to handle activity logging, streak tracking, and retrieving historical activity.
- **A SwiftUI Activity View** that allows users to input new activities and visualize past logged activities.
- **Unit tests** to validate functionality, ensuring correctness in streak calculation and data management.

---

### **🚀 Features Added**
✔ **Activity Model (`Activity.swift`)**
- Stores details like **steps, active minutes, calories burned, and activity type**.
- Includes **computed properties** to dynamically translate steps into active minutes.
- Provides a **helper function (`toDictionary()`)** to serialize data for Firebase storage.

✔ **Activity Manager (`ActivityManager.swift`)**
- **Logs activities** and maintains an in-memory activity list.
- Computes **streaks dynamically** based on consecutive days meeting the 60-minute goal.
- Provides a **motivational feedback system** based on user progress.
- Implements **data persistence** using `UserDefaults` (temporary storage before integrating Firebase).

✔ **Activity View (`ActivityView.swift`)**
- **Allows users to input and log activities** through a simple form.
- **Displays a history of logged activities** with dynamically updated streak information.
- **Uses clean UI formatting** to enhance readability and engagement.

✔ **Unit Tests (`ActivityManagerTests.swift`)**
- **Ensures streak calculation works correctly** under different scenarios.
- Validates **retrieving today's activity** and **logging new activities**.
- **Uses mock data** to simulate user activity.

---

## **📚 Documentation**
This PR follows the **[[Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md)](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md)**.

### **💡 Key Implementation Notes**
- **Activity Model (`Activity.swift`)**
  - Designed to store relevant activity data for kids.
  - Uses a **computed property to translate steps into active minutes** instead of storing redundant data.

- **Activity Manager (`ActivityManager.swift`)**
  - `checkStreak()` **computes the streak dynamically** rather than storing it in the `Activity` model.
  - `triggerMotivation()` **provides feedback messages** based on progress.

- **Activity View (`ActivityView.swift`)**
  - Extracted `ActivityRow` for **modular and reusable UI components**.
  - Improved **form styling and UI layout** for better aesthetics.
  - **Added preview support** with mock data.

---

### **Test Cases Added (`ActivityManagerTests.swift`)**
✔ **Test Streak Calculation (`testCheckStreak()`)**  
- Verifies that streaks are correctly computed when activities meet the 60-minute threshold.  

✔ **Test Fetching Today’s Activity (`testGetTodayActivity()`)**  
- Ensures the function correctly identifies and retrieves today's activity.  

✔ **Test Logging a New Activity (`testLogActivity()`)**  
- Confirms that activities are properly appended and stored.  

### **🚨 Edge Cases Covered**
✔ **Break in Streaks:** Ensures that missing a day resets the streak.  
✔ **Different Activity Durations:** Tests handling of activities below and above the 60-minute threshold.  
✔ **Non-Consecutive Days:** Ensures that skipping a day prevents streak accumulation.

---

## **📝 [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) & Contributing Guidelines**
By submitting this pull request, you confirm that you adhere to the **[Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md)** and **[[Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md)](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md)**.

- [x] I agree to follow the **Code of Conduct**.
- [x] I have written **unit tests** for new features.
- [x] I have **documented my code** properly in accordance with the Spezi guidelines.

---

## **📌 Next Steps**
- **Integrate Firebase storage** instead of `UserDefaults` for persistent data.
- **Connect with Apple HealthKit** for automatic activity tracking.
- **Enhance UI** with activity completion progress bars or animations.
- **Increase Test Coverage**.

### CODE REVIEW

File: Stanford360.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -276,6 +299,8 @@
 		653A254F283387FE005D4D48 /* Stanford360 */ = {
 			isa = PBXGroup;
 			children = (
+				4FA061CF2D4C4468008DE21A /* Views */,
+				4FA061C62D4C43A7008DE21A /* Drafts */,
```
Comment (kkellybonilla) [inline Stanford360.xcodeproj/project.pbxproj]: delete this line

File: Stanford360.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -354,6 +381,10 @@
 				A9E1D3432C67A3F800CED217 /* PBXTargetDependency */,
 				56E7083D2BB06FCA00B08F0A /* PBXTargetDependency */,
 			);
+			fileSystemSynchronizedGroups = (
+				4FA061C62D4C43A7008DE21A /* Drafts */,
```
Comment (kkellybonilla) [inline Stanford360.xcodeproj/project.pbxproj]: delete this line

File: Stanford360/Models/Activity.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+
+import Foundation
+
+/// Represents a child's daily physical activity session.
+struct Activity: Identifiable, Codable {
+    var id: UUID = UUID()
```
Comment (kkellybonilla) [inline Stanford360/Models/Activity.swift]: initialize the id in the constructor, make this just `var id: UUID` so that the error about "redundant type annotation" goes away

File: Stanford360/Models/Activity.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    var userID: UUID
+    var date: Date
+    var steps: Int
+    var activeMinutes: Int
+    var caloriesBurned: Int
+    var activityType: String
+    
+    init(id: UUID, userID: UUID, date: Date, steps: Int, activeMinutes: Int, caloriesBurned: Int, activityType: String) {
+        self.id = id
```
Comment (kkellybonilla) [inline Stanford360/Models/Activity.swift]: self.id = UUID()

**Review by kkellybonilla — CHANGES_REQUESTED** [review]

**Review by kkellybonilla — APPROVED** [review]
